File: breath_trainer.go
# breath_trainer.go
/**
 * 🌬️ Breath Trainer – 4-7-8 Relaxation Technique (Go Edition)
 * Features: guided breathing, progress bars, stats, custom timing, sound
 */

package main

import (
	"bufio"
	"encoding/json"
	"fmt"
	"os"
	"path/filepath"
	"strconv"
	"strings"
	"time"
)

// ─── Types ──────────────────────────────────────────────────────────────────

type Timing struct {
	Inhale int `json:"inhale"`
	Hold   int `json:"hold"`
	Exhale int `json:"exhale"`
}

type Session struct {
	Date            string `json:"date"`
	Cycles          int    `json:"cycles"`
	DurationSeconds int    `json:"durationSeconds"`
}

type Stats struct {
	TotalSessions   int       `json:"totalSessions"`
	TotalCycles     int       `json:"totalCycles"`
	TotalTimeSeconds int      `json:"totalTimeSeconds"`
	Sessions        []Session `json:"sessions"`
	CustomTiming    Timing    `json:"customTiming"`
}

// ─── Colors ──────────────────────────────────────────────────────────────────

const (
	reset  = "\x1b[0m"
	bright = "\x1b[1m"
	dim    = "\x1b[2m"
	red    = "\x1b[31m"
	green  = "\x1b[32m"
	yellow = "\x1b[33m"
	blue   = "\x1b[34m"
	magenta = "\x1b[35m"
	cyan   = "\x1b[36m"
)

func c(str, color string) string {
	return color + str + reset
}

// ─── Sound ──────────────────────────────────────────────────────────────────

func beep() {
	fmt.Print("\x07")
}

// ─── Config ──────────────────────────────────────────────────────────────────

const (
	defaultInhale = 4
	defaultHold   = 7
	defaultExhale = 8
)

// ─── Stats Manager ──────────────────────────────────────────────────────────

type StatsManager struct {
	filePath        string
	TotalSessions   int
	TotalCycles     int
	TotalTimeSeconds int
	Sessions        []Session
	CustomTiming    Timing
}

func NewStatsManager() *StatsManager {
	home, _ := os.UserHomeDir()
	dir := filepath.Join(home, ".breath_trainer")
	os.MkdirAll(dir, 0755)
	filePath := filepath.Join(dir, "stats.json")
	s := &StatsManager{filePath: filePath}
	s.load()
	return s
}

func (s *StatsManager) load() {
	if _, err := os.Stat(s.filePath); os.IsNotExist(err) {
		s.CustomTiming = Timing{Inhale: defaultInhale, Hold: defaultHold, Exhale: defaultExhale}
		return
	}
	raw, err := os.ReadFile(s.filePath)
	if err != nil {
		s.CustomTiming = Timing{Inhale: defaultInhale, Hold: defaultHold, Exhale: defaultExhale}
		return
	}
	var data Stats
	if err := json.Unmarshal(raw, &data); err != nil {
		s.CustomTiming = Timing{Inhale: defaultInhale, Hold: defaultHold, Exhale: defaultExhale}
		return
	}
	s.TotalSessions = data.TotalSessions
	s.TotalCycles = data.TotalCycles
	s.TotalTimeSeconds = data.TotalTimeSeconds
	s.Sessions = data.Sessions
	s.CustomTiming = data.CustomTiming
	if s.CustomTiming.Inhale == 0 {
		s.CustomTiming = Timing{Inhale: defaultInhale, Hold: defaultHold, Exhale: defaultExhale}
	}
}

func (s *StatsManager) save() {
	data := Stats{
		TotalSessions:    s.TotalSessions,
		TotalCycles:      s.TotalCycles,
		TotalTimeSeconds: s.TotalTimeSeconds,
		Sessions:         s.Sessions,
		CustomTiming:     s.CustomTiming,
	}
	raw, _ := json.MarshalIndent(data, "", "  ")
	os.WriteFile(s.filePath, raw, 0644)
}

func (s *StatsManager) RecordSession(cycles int, duration float64) {
	s.TotalSessions++
	s.TotalCycles += cycles
	s.TotalTimeSeconds += int(duration)
	s.Sessions = append(s.Sessions, Session{
		Date:            time.Now().Format(time.RFC3339),
		Cycles:          cycles,
		DurationSeconds: int(duration),
	})
	if len(s.Sessions) > 50 {
		s.Sessions = s.Sessions[len(s.Sessions)-50:]
	}
	s.save()
}

func (s *StatsManager) SetTiming(inhale, hold, exhale int) {
	s.CustomTiming = Timing{Inhale: inhale, Hold: hold, Exhale: exhale}
	s.save()
}

// ─── Breath Trainer ──────────────────────────────────────────────────────

type BreathTrainer struct {
	reader   *bufio.Reader
	stats    *StatsManager
	inhale   int
	hold     int
	exhale   int
	soundOn  bool
}

func NewBreathTrainer() *BreathTrainer {
	s := NewStatsManager()
	return &BreathTrainer{
		reader:  bufio.NewReader(os.Stdin),
		stats:   s,
		inhale:  s.CustomTiming.Inhale,
		hold:    s.CustomTiming.Hold,
		exhale:  s.CustomTiming.Exhale,
		soundOn: true,
	}
}

func (b *BreathTrainer) sleep(ms int) {
	time.Sleep(time.Duration(ms) * time.Millisecond)
}

func (b *BreathTrainer) showStage(stage string, duration int, color, emoji string) {
	steps := duration * 2
	for i := 0; i <= steps; i++ {
		pct := float64(i) / float64(steps) * 100
		filled := int(float64(i) / float64(steps) * 30)
		bar := strings.Repeat("█", filled) + strings.Repeat("░", 30-filled)
		fmt.Printf("\r  %s %s %s %3.0f%%", emoji, c(stage, color), c(bar, color), pct)
		b.sleep(500)
	}
	fmt.Println()
}

func (b *BreathTrainer) breathingCycle() float64 {
	start := time.Now()
	if b.soundOn {
		beep()
	}
	b.showStage("Breathe In...", b.inhale, blue, "🌬️")
	if b.soundOn {
		beep()
	}
	b.showStage("Hold...", b.hold, yellow, "⏸️")
	if b.soundOn {
		beep()
	}
	b.showStage("Exhale...", b.exhale, green, "🌊")
	return time.Since(start).Seconds()
}

func (b *BreathTrainer) RunSession(cycles int) {
	fmt.Printf("%s\n", c(fmt.Sprintf("\n🌬️ 4-7-8 Breathing Session – %d cycles", cycles), bright+cyan))
	if b.soundOn {
		beep()
	}
	fmt.Println(c("Get ready...", dim))
	b.sleep(1000)

	var totalDuration float64
	for i := 0; i < cycles; i++ {
		fmt.Printf("%s\n", c(fmt.Sprintf("\nCycle %d/%d", i+1, cycles), magenta))
		totalDuration += b.breathingCycle()
	}

	if b.soundOn {
		beep()
		beep()
		beep()
	}
	fmt.Printf("%s\n", c("\n✨ Session complete! ✨", green))
	fmt.Printf("  Cycles: %d\n", cycles)
	fmt.Printf("  Duration: %.1f seconds\n", totalDuration)
	b.stats.RecordSession(cycles, totalDuration)
}

func (b *BreathTrainer) ShowStats() {
	fmt.Println("\n📊 STATISTICS")
	fmt.Println(c(strings.Repeat("─", 30), dim))
	fmt.Printf("  Total Sessions: %d\n", b.stats.TotalSessions)
	fmt.Printf("  Total Cycles:   %d\n", b.stats.TotalCycles)
	mins := b.stats.TotalTimeSeconds / 60
	secs := b.stats.TotalTimeSeconds % 60
	fmt.Printf("  Total Time:     %dm %ds\n", mins, secs)
	if len(b.stats.Sessions) > 0 {
		last := b.stats.Sessions[len(b.stats.Sessions)-1]
		fmt.Printf("  Last Session:   %d cycles (%ds)\n", last.Cycles, last.DurationSeconds)
		fmt.Println("\n📅 Recent Sessions:")
		start := 0
		if len(b.stats.Sessions) > 5 {
			start = len(b.stats.Sessions) - 5
		}
		for _, s := range b.stats.Sessions[start:] {
			d := s.Date[:16]
			d = strings.Replace(d, "T", " ", 1)
			fmt.Printf("  %s  %d cycles  %ds\n", d, s.Cycles, s.DurationSeconds)
		}
	}
}

func (b *BreathTrainer) SetTiming() {
	inhale := b.askInt(fmt.Sprintf("Inhale duration (default %d): ", b.inhale))
	hold := b.askInt(fmt.Sprintf("Hold duration (default %d): ", b.hold))
	exhale := b.askInt(fmt.Sprintf("Exhale duration (default %d): ", b.exhale))
	if inhale > 0 {
		b.inhale = inhale
	}
	if hold > 0 {
		b.hold = hold
	}
	if exhale > 0 {
		b.exhale = exhale
	}
	b.stats.SetTiming(b.inhale, b.hold, b.exhale)
	fmt.Printf("%s\n", c(fmt.Sprintf("✅ Timing set: %ds inhale, %ds hold, %ds exhale", b.inhale, b.hold, b.exhale), green))
}

func (b *BreathTrainer) ToggleSound() {
	b.soundOn = !b.soundOn
	state := "on"
	if !b.soundOn {
		state = "off"
	}
	fmt.Printf("%s\n", c(fmt.Sprintf("🔊 Sound %s", state), cyan))
}

func (b *BreathTrainer) ResetStats() {
	fmt.Print("⚠️  Reset all statistics? (yes/no): ")
	ans, _ := b.reader.ReadString('\n')
	ans = strings.TrimSpace(strings.ToLower(ans))
	if ans == "yes" {
		b.stats.TotalSessions = 0
		b.stats.TotalCycles = 0
		b.stats.TotalTimeSeconds = 0
		b.stats.Sessions = []Session{}
		b.stats.save()
		fmt.Printf("%s\n", c("🗑️  Statistics reset.", yellow))
	}
}

// ─── Helpers ──────────────────────────────────────────────────────────────

func (b *BreathTrainer) ask(prompt string) string {
	fmt.Print(prompt)
	line, _ := b.reader.ReadString('\n')
	return strings.TrimSpace(line)
}

func (b *BreathTrainer) askInt(prompt string) int {
	for {
		ans := b.ask(prompt)
		if ans == "" {
			return 0
		}
		if val, err := strconv.Atoi(ans); err == nil {
			return val
		}
		fmt.Println(c("❌ Please enter a number.", red))
	}
}

func (b *BreathTrainer) showMenu() {
	fmt.Println("\n" + c(strings.Repeat("═", 50), cyan))
	fmt.Println(c("🌬️ BREATH TRAINER", bright+cyan))
	fmt.Println(c(strings.Repeat("═", 50), cyan))
	fmt.Printf("  4-7-8: Inhale %ds, Hold %ds, Exhale %ds\n", b.inhale, b.hold, b.exhale)
	state := "🔊 On"
	if !b.soundOn {
		state = "🔇 Off"
	}
	fmt.Printf("  Sound: %s\n", state)
	fmt.Println(c(strings.Repeat("═", 50), cyan))
	fmt.Println("  1. 🌬️ Start Session (4 cycles)")
	fmt.Println("  2. 🌬️ Custom Session (choose cycles)")
	fmt.Println("  3. 📊 Statistics")
	fmt.Println("  4. ⏱️ Set Timings")
	fmt.Println("  5. 🔇 Toggle Sound")
	fmt.Println("  6. 🗑️ Reset Statistics")
	fmt.Println("  0. 🚪 Exit")
	fmt.Println(c(strings.Repeat("═", 50), cyan))
}

func (b *BreathTrainer) run() {
	fmt.Print("\033[H\033[2J")
	fmt.Printf("%s\n", c("\n🌬️ Breath Trainer – 4-7-8 Relaxation", bright+cyan))
	fmt.Printf("%s\n", c("Breathe in peace, hold calm, exhale stress.", dim))

	for {
		b.showMenu()
		choice := b.ask("Your choice: ")
		switch choice {
		case "1":
			b.RunSession(4)
		case "2":
			cycles := b.askInt("Number of cycles (default 4): ")
			if cycles <= 0 {
				cycles = 4
			}
			b.RunSession(cycles)
		case "3":
			b.ShowStats()
		case "4":
			b.SetTiming()
		case "5":
			b.ToggleSound()
		case "6":
			b.ResetStats()
		case "0":
			fmt.Printf("%s\n", c("👋 Breathe deeply! Goodbye!", cyan))
			return
		default:
			fmt.Println(c("❌ Invalid choice.", red))
		}
		if choice != "0" {
			fmt.Print("\nPress Enter to continue...")
			b.reader.ReadString('\n')
		}
	}
}

func main() {
	app := NewBreathTrainer()
	app.run()
}

File: breath_trainer.rs
# breath_trainer.rs
/**
 * 🌬️ Breath Trainer – 4-7-8 Relaxation Technique (Rust Edition)
 * Features: guided breathing, progress bars, stats, custom timing, sound
 * Dependencies: serde, serde_json, chrono, colored, crossterm
 */

use chrono::Utc;
use colored::*;
use crossterm::terminal;
use serde::{Deserialize, Serialize};
use std::fs;
use std::io::{self, Write, BufRead};
use std::path::PathBuf;
use std::thread;
use std::time::Duration;

// ─── Types ──────────────────────────────────────────────────────────────────

#[derive(Debug, Serialize, Deserialize, Clone)]
struct Timing {
    inhale: u32,
    hold: u32,
    exhale: u32,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
struct Session {
    date: String,
    cycles: u32,
    duration_seconds: u32,
}

#[derive(Debug, Serialize, Deserialize)]
struct Stats {
    total_sessions: u32,
    total_cycles: u32,
    total_time_seconds: u32,
    sessions: Vec<Session>,
    custom_timing: Timing,
}

// ─── Colors ──────────────────────────────────────────────────────────────────

fn c(text: &str, color: &str) -> String {
    match color {
        "green" => text.green().to_string(),
        "red" => text.red().to_string(),
        "yellow" => text.yellow().to_string(),
        "cyan" => text.cyan().to_string(),
        "magenta" => text.magenta().to_string(),
        "blue" => text.blue().to_string(),
        "bright" => text.bright().to_string(),
        "dim" => text.dimmed().to_string(),
        _ => text.to_string(),
    }
}

// ─── Sound ──────────────────────────────────────────────────────────────────

fn beep() {
    print!("\x07");
    io::stdout().flush().unwrap();
}

// ─── Config ──────────────────────────────────────────────────────────────────

const DEFAULT_INHALE: u32 = 4;
const DEFAULT_HOLD: u32 = 7;
const DEFAULT_EXHALE: u32 = 8;

// ─── Stats Manager ──────────────────────────────────────────────────────────

struct StatsManager {
    file_path: PathBuf,
    pub total_sessions: u32,
    pub total_cycles: u32,
    pub total_time_seconds: u32,
    pub sessions: Vec<Session>,
    pub custom_timing: Timing,
}

impl StatsManager {
    fn new() -> Self {
        let home = std::env::var("HOME").or_else(|_| std::env::var("USERPROFILE")).unwrap_or_else(|_| ".".to_string());
        let dir = PathBuf::from(home).join(".breath_trainer");
        fs::create_dir_all(&dir).unwrap();
        let file_path = dir.join("stats.json");
        let mut s = StatsManager {
            file_path,
            total_sessions: 0,
            total_cycles: 0,
            total_time_seconds: 0,
            sessions: Vec::new(),
            custom_timing: Timing { inhale: DEFAULT_INHALE, hold: DEFAULT_HOLD, exhale: DEFAULT_EXHALE },
        };
        s.load();
        s
    }

    fn load(&mut self) {
        if let Ok(raw) = fs::read_to_string(&self.file_path) {
            if let Ok(data) = serde_json::from_str::<Stats>(&raw) {
                self.total_sessions = data.total_sessions;
                self.total_cycles = data.total_cycles;
                self.total_time_seconds = data.total_time_seconds;
                self.sessions = data.sessions;
                self.custom_timing = data.custom_timing;
                if self.custom_timing.inhale == 0 {
                    self.custom_timing = Timing { inhale: DEFAULT_INHALE, hold: DEFAULT_HOLD, exhale: DEFAULT_EXHALE };
                }
                return;
            }
        }
        self.custom_timing = Timing { inhale: DEFAULT_INHALE, hold: DEFAULT_HOLD, exhale: DEFAULT_EXHALE };
    }

    fn save(&self) {
        let data = Stats {
            total_sessions: self.total_sessions,
            total_cycles: self.total_cycles,
            total_time_seconds: self.total_time_seconds,
            sessions: self.sessions.clone(),
            custom_timing: self.custom_timing.clone(),
        };
        let raw = serde_json::to_string_pretty(&data).unwrap();
        let _ = fs::write(&self.file_path, raw);
    }

    fn record_session(&mut self, cycles: u32, duration: f64) {
        self.total_sessions += 1;
        self.total_cycles += cycles;
        self.total_time_seconds += duration as u32;
        self.sessions.push(Session {
            date: Utc::now().to_rfc3339(),
            cycles,
            duration_seconds: duration as u32,
        });
        if self.sessions.len() > 50 {
            self.sessions.drain(0..self.sessions.len()-50);
        }
        self.save();
    }

    fn set_timing(&mut self, inhale: u32, hold: u32, exhale: u32) {
        self.custom_timing = Timing { inhale, hold, exhale };
        self.save();
    }
}

// ─── Breath Trainer ──────────────────────────────────────────────────────

struct BreathTrainer {
    stats: StatsManager,
    inhale: u32,
    hold: u32,
    exhale: u32,
    sound_on: bool,
}

impl BreathTrainer {
    fn new() -> Self {
        let stats = StatsManager::new();
        let inhale = stats.custom_timing.inhale;
        let hold = stats.custom_timing.hold;
        let exhale = stats.custom_timing.exhale;
        BreathTrainer { stats, inhale, hold, exhale, sound_on: true }
    }

    fn sleep_ms(&self, ms: u64) {
        thread::sleep(Duration::from_millis(ms));
    }

    fn show_stage(&self, stage: &str, duration: u32, color: &str, emoji: &str) {
        let steps = duration * 2;
        for i in 0..=steps {
            let pct = i as f64 / steps as f64 * 100.0;
            let filled = (i as f64 / steps as f64 * 30.0) as usize;
            let bar = "█".repeat(filled) + &"░".repeat(30 - filled);
            print!("\r  {} {} {} {:3.0}%", emoji, c(stage, color), c(&bar, color), pct);
            io::stdout().flush().unwrap();
            self.sleep_ms(500);
        }
        println!();
    }

    fn breathing_cycle(&self) -> f64 {
        let start = std::time::Instant::now();
        if self.sound_on { beep(); }
        self.show_stage("Breathe In...", self.inhale, "blue", "🌬️");
        if self.sound_on { beep(); }
        self.show_stage("Hold...", self.hold, "yellow", "⏸️");
        if self.sound_on { beep(); }
        self.show_stage("Exhale...", self.exhale, "green", "🌊");
        start.elapsed().as_secs_f64()
    }

    fn run_session(&mut self, cycles: u32) {
        println!("{}", c(&format!("\n🌬️ 4-7-8 Breathing Session – {} cycles", cycles), "bright cyan"));
        if self.sound_on { beep(); }
        println!("{}", c("Get ready...", "dim"));
        self.sleep_ms(1000);

        let mut total_duration = 0.0;
        for i in 0..cycles {
            println!("{}", c(&format!("\nCycle {}/{}", i+1, cycles), "magenta"));
            total_duration += self.breathing_cycle();
        }

        if self.sound_on { beep(); beep(); beep(); }
        println!("{}", c("\n✨ Session complete! ✨", "green"));
        println!("  Cycles: {}", cycles);
        println!("  Duration: {:.1} seconds", total_duration);
        self.stats.record_session(cycles, total_duration);
    }

    fn show_stats(&self) {
        println!("\n📊 STATISTICS");
        println!("{}", "─".repeat(30).dimmed());
        println!("  Total Sessions: {}", self.stats.total_sessions);
        println!("  Total Cycles:   {}", self.stats.total_cycles);
        let mins = self.stats.total_time_seconds / 60;
        let secs = self.stats.total_time_seconds % 60;
        println!("  Total Time:     {}m {}s", mins, secs);
        if !self.stats.sessions.is_empty() {
            let last = &self.stats.sessions[self.stats.sessions.len()-1];
            println!("  Last Session:   {} cycles ({}s)", last.cycles, last.duration_seconds);
            println!("\n📅 Recent Sessions:");
            let start = if self.stats.sessions.len() > 5 { self.stats.sessions.len() - 5 } else { 0 };
            for s in &self.stats.sessions[start..] {
                let d = s.date[..16].replace("T", " ");
                println!("  {}  {} cycles  {}s", d, s.cycles, s.duration_seconds);
            }
        }
    }

    fn set_timing(&mut self) {
        let inhale = self.ask_u32(&format!("Inhale duration (default {}): ", self.inhale));
        let hold = self.ask_u32(&format!("Hold duration (default {}): ", self.hold));
        let exhale = self.ask_u32(&format!("Exhale duration (default {}): ", self.exhale));
        if inhale > 0 { self.inhale = inhale; }
        if hold > 0 { self.hold = hold; }
        if exhale > 0 { self.exhale = exhale; }
        self.stats.set_timing(self.inhale, self.hold, self.exhale);
        println!("{}", c(&format!("✅ Timing set: {}s inhale, {}s hold, {}s exhale", self.inhale, self.hold, self.exhale), "green"));
    }

    fn toggle_sound(&mut self) {
        self.sound_on = !self.sound_on;
        let state = if self.sound_on { "on" } else { "off" };
        println!("{}", c(&format!("🔊 Sound {}", state), "cyan"));
    }

    fn reset_stats(&mut self) {
        print!("⚠️  Reset all statistics? (yes/no): ");
        io::stdout().flush().unwrap();
        let mut ans = String::new();
        io::stdin().read_line(&mut ans).unwrap();
        if ans.trim().to_lowercase() == "yes" {
            self.stats.total_sessions = 0;
            self.stats.total_cycles = 0;
            self.stats.total_time_seconds = 0;
            self.stats.sessions = Vec::new();
            self.stats.save();
            println!("{}", c("🗑️  Statistics reset.", "yellow"));
        }
    }

    // ─── Helpers ──────────────────────────────────────────────────────────

    fn ask(&self, prompt: &str) -> String {
        print!("{}", prompt);
        io::stdout().flush().unwrap();
        let mut line = String::new();
        io::stdin().read_line(&mut line).unwrap();
        line.trim().to_string()
    }

    fn ask_u32(&self, prompt: &str) -> u32 {
        loop {
            let ans = self.ask(prompt);
            if ans.is_empty() { return 0; }
            if let Ok(val) = ans.parse::<u32>() {
                return val;
            }
            println!("{}", c("❌ Please enter a number.", "red"));
        }
    }

    fn show_menu(&self) {
        let state = if self.sound_on { "🔊 On" } else { "🔇 Off" };
        println!("\n{}", "═".repeat(50).cyan());
        println!("{}", c("🌬️ BREATH TRAINER", "bright cyan"));
        println!("{}", "═".repeat(50).cyan());
        println!("  4-7-8: Inhale {}s, Hold {}s, Exhale {}s", self.inhale, self.hold, self.exhale);
        println!("  Sound: {}", state);
        println!("{}", "═".repeat(50).cyan());
        println!("  1. 🌬️ Start Session (4 cycles)");
        println!("  2. 🌬️ Custom Session (choose cycles)");
        println!("  3. 📊 Statistics");
        println!("  4. ⏱️ Set Timings");
        println!("  5. 🔇 Toggle Sound");
        println!("  6. 🗑️ Reset Statistics");
        println!("  0. 🚪 Exit");
        println!("{}", "═".repeat(50).cyan());
    }

    fn run(&mut self) {
        println!("{}", "\n🌬️ Breath Trainer – 4-7-8 Relaxation".bright().cyan());
        println!("{}", "Breathe in peace, hold calm, exhale stress.".dimmed());

        loop {
            self.show_menu();
            let choice = self.ask("Your choice: ");
            match choice.as_str() {
                "1" => self.run_session(4),
                "2" => {
                    let cycles = self.ask_u32("Number of cycles (default 4): ");
                    let cycles = if cycles == 0 { 4 } else { cycles };
                    self.run_session(cycles);
                }
                "3" => self.show_stats(),
                "4" => self.set_timing(),
                "5" => self.toggle_sound(),
                "6" => self.reset_stats(),
                "0" => {
                    println!("{}", c("👋 Breathe deeply! Goodbye!", "cyan"));
                    return;
                }
                _ => println!("{}", c("❌ Invalid choice.", "red")),
            }
            if choice != "0" {
                print!("\nPress Enter to continue...");
                io::stdout().flush().unwrap();
                let mut _dummy = String::new();
                io::stdin().read_line(&mut _dummy).unwrap();
            }
        }
    }
}

// ─── Main ────────────────────────────────────────────────────────────────────

fn main() {
    let mut app = BreathTrainer::new();
    app.run();
}
